common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp

```diff
@@ -11,5 +14,51 @@ Point RNSScreenShadowNode::getContentOriginOffset() const {
   return {contentOffset.x, contentOffset.y};
 }
 
+void RNSScreenShadowNode::layout(facebook::react::LayoutContext layoutContext) {
+  YogaLayoutableShadowNode::layout(layoutContext);
+
+#ifdef ANDROID
+  applyFrameCorrections();
+#endif // ANDROID
+}
+
+#ifdef ANDROID
+void RNSScreenShadowNode::applyFrameCorrections() {
+  ensureUnsealed();
+
+  // On the very first layout we want to correct both Y offset and frame size.
```

Comment:
So here we try to mimic the values that will be eventually set on the shadow node so that there is no new layout passes after they come? Why is this needed here too and how does it differ from what we do in `adopt`?

Reply:
It's because `layoutMetrics_` are ignored and `reanimated` reads from them?

Reply:
In adopt we apply padding, so that during layout pass `Yoga` layouts `Screen`'s children with less available space. `Yoga` won't modify `Screen`'s size / frame origin itself. 
Here we do this manually, so when (e.g. reanimated) someone tries to read `Screen` dimensions from the shadow node they are possible accurate. If someone tries to hack around and read from `yogaNode_` private field, he will get the wrong value, but we don't care, if someone does this he's already doing something wrong.

Reply:
> It's because layoutMetrics_ are ignored and reanimated reads from them?

Yeah, changes to `layoutMetrics_` do not trigger layout pass. This is just a simple data field w/o any logic attached to setters / getters. ScrollView does the same. 

Also, this field is later read when sending updates to UI.